### DISCUSSION TITLE
E3372 does not work with 2022.45.1

### DISCUSSION THREAD

**Discussion by sacha81** [tesla-android/issue-tracker#167] [General]
I just started with Tesla-Android as I was waiting months to get a Rpi4. Now finally it works, it works really great! Awesome Michael!
But I don't have Internet access, I have plugged in a E3372 with a working SIM card. It flashes blue, what imho means that its connected. Do I miss something or has 2022.45.1 problems with the Huawei E3372?

**Comment by mikegapinski** [comment]
It should be solid blue... 

Did you plug it in before turning the pi on?

**Reply by sacha81** [reply]
Yes, plugged in before power on.

**Reply by mikegapinski** [reply]
@sacha81 Can you plug the pi to a computer with adb installed and send me the output of a few commands. Upload them to pastebin if they are long! 

adb shell logcat
adb shell dmesg
adb shell ifconfig -a

I should be able to help quickly once I get that from you

**Reply by Petto16** [reply]
HI, I have the same issue with E3372-153, but with E3372-320 it is working fine.

**Reply by mikegapinski** [reply]
@Petto16 Maybe your E3372-153 is not in hi link model like the -320 version. Can you share the lsusb and ifconfig from both?

**Reply by Petto16** [reply]
E3372h-153
Bus 003 Device 001: ID 1d6b:0002
Bus 001 Device 001: ID 1d6b:0002
Bus 001 Device 004: ID 12d1:14fe
Bus 001 Device 002: ID 2109:3431
Bus 002 Device 001: ID 1d6b:0003
Bus 001 Device 009: ID 1314:1521

lo        Link encap:Local Loopback  
          inet addr:127.0.0.1  Mask:255.0.0.0 
          inet6 addr: ::1/128 Scope: Host
          UP LOOPBACK RUNNING  MTU:65536  Metric:1
          RX packets:0 errors:0 dropped:0 overruns:0 frame:0 
          TX packets:0 errors:0 dropped:0 overruns:0 carrier:0 
          collisions:0 txqueuelen:1000 
          RX bytes:0 TX bytes:0 

dummy0    Link encap:Ethernet  HWaddr 7a:59:bd:38:7e:9d
          inet6 addr: fe80::7859:bdff:fe38:7e9d/64 Scope: Link
          UP BROADCAST RUNNING NOARP  MTU:1500  Metric:1
          RX packets:0 errors:0 dropped:0 overruns:0 frame:0 
          TX packets:7 errors:0 dropped:0 overruns:0 carrier:0 
          collisions:0 txqueuelen:1000 
          RX bytes:0 TX bytes:490 

eth0      Link encap:Ethernet  HWaddr dc:a6:32:65:b2:57  Driver bcmgenet
          UP BROADCAST MULTICAST  MTU:1500  Metric:1
          RX packets:0 errors:0 dropped:0 overruns:0 frame:0 
          TX packets:0 errors:0 dropped:0 overruns:0 carrier:0 
          collisions:0 txqueuelen:1000 
          RX bytes:0 TX bytes:0 

wlan0     Link encap:Ethernet  HWaddr dc:a6:32:65:b2:58  Driver brcmfmac
          inet addr:9.9.0.1  Bcast:9.9.0.255  Mask:255.255.255.0 
          inet6 addr: fe80::dea6:32ff:fe65:b258/64 Scope: Link
          UP BROADCAST RUNNING MULTICAST  MTU:1500  Metric:1
          RX packets:0 errors:0 dropped:0 overruns:0 frame:0 
          TX packets:8 errors:0 dropped:0 overruns:0 carrier:0 
          collisions:0 txqueuelen:1000 
          RX bytes:0 TX bytes:1048

**Reply by Petto16** [reply]
E3372h-320

Bus 003 Device 001: ID 1d6b:0002
Bus 001 Device 001: ID 1d6b:0002
Bus 001 Device 009: ID 12d1:14db
Bus 001 Device 002: ID 2109:3431
Bus 002 Device 001: ID 1d6b:0003
Bus 001 Device 010: ID 1314:1521

lo        Link encap:Local Loopback  
          inet addr:127.0.0.1  Mask:255.0.0.0 
          inet6 addr: ::1/128 Scope: Host
          UP LOOPBACK RUNNING  MTU:65536  Metric:1
          RX packets:0 errors:0 dropped:0 overruns:0 frame:0 
          TX packets:0 errors:0 dropped:0 overruns:0 carrier:0 
          collisions:0 txqueuelen:1000 
          RX bytes:0 TX bytes:0 

dummy0    Link encap:Ethernet  HWaddr 0a:a6:a3:62:12:ff
          inet6 addr: fe80::8a6:a3ff:fe62:12ff/64 Scope: Link
          UP BROADCAST RUNNING NOARP  MTU:1500  Metric:1
          RX packets:0 errors:0 dropped:0 overruns:0 frame:0 
          TX packets:5 errors:0 dropped:0 overruns:0 carrier:0 
          collisions:0 txqueuelen:1000 
          RX bytes:0 TX bytes:350 

eth0      Link encap:Ethernet  HWaddr dc:a6:32:65:b2:57  Driver bcmgenet
          UP BROADCAST MULTICAST  MTU:1500  Metric:1
          RX packets:0 errors:0 dropped:0 overruns:0 frame:0 
          TX packets:0 errors:0 dropped:0 overruns:0 carrier:0 
          collisions:0 txqueuelen:1000 
          RX bytes:0 TX bytes:0 

wlan0     Link encap:Ethernet  HWaddr dc:a6:32:65:b2:58  Driver brcmfmac
          inet addr:9.9.0.1  Bcast:9.9.0.255  Mask:255.255.255.0 
          inet6 addr: fe80::dea6:32ff:fe65:b258/64 Scope: Link
          UP BROADCAST RUNNING MULTICAST  MTU:1500  Metric:1
          RX packets:0 errors:0 dropped:0 overruns:0 frame:0 
          TX packets:9 errors:0 dropped:0 overruns:0 carrier:0 
          collisions:0 txqueuelen:1000 
          RX bytes:0 TX bytes:1164 

eth1      Link encap:Ethernet  HWaddr 00:1e:10:1f:00:00  Driver cdc_ether
          inet addr:192.168.8.102  Bcast:192.168.8.255  Mask:255.255.255.0 
          inet6 addr: fe80::427f:c3ef:b057:8747/64 Scope: Link
          UP BROADCAST RUNNING MULTICAST  MTU:1500  Metric:1
          RX packets:47868 errors:0 dropped:0 overruns:0 frame:0 
          TX packets:7822 errors:0 dropped:0 overruns:0 carrier:0 
          collisions:0 txqueuelen:1000 
          RX bytes:61267809 TX bytes:780561

**Reply by Petto16** [reply]
[logcat 320 and 153.zip](https://github.com/tesla-android/issue-tracker/files/10076834/logcat.320.and.153.zip)

**Reply by Petto16** [reply]
@mikegapinski how I can change it to hilink mode?

**Reply by mikegapinski** [reply]
usb modeswitch I guess, I would add the config for your variant with a next update. Thanks for the logs!

**Reply by sacha81** [reply]
same here, E3372-153 verison. So I guess the logs from me are obsolete.

**Comment by mikegapinski** [comment]
@Petto16 @sacha81 I've found a tutorial on how you can convert the modem to hi link mode:

https://gist.github.com/smarek/cbe2ae6f9bcd76d610c013668df880e5

Anyone brave enough to give this a shot? It's a firmware change, if it'll work there won't be any need for me to change anything in Tesla Android...

**Reply by sacha81** [reply]
I had a look on this instructions, but I'll try to get a E3372h-320 now. Wont brake my E3372-153. Thanks @mikegapinski for the research. As we are not the only one with this problem, maybe a support for E3372h-153 makes sense.

**Reply by mikegapinski** [reply]
I would rather have you guys change the firmware since it's not rocket science or anything. Support for your variant might come one day. BTW: https://github.com/tesla-android/android-manifest/releases/tag/2022.45.1.1-modem-test check this image, and see if it changes anything with the commands I sent earlier

**Reply by Petto16** [reply]
@mikegapinski Just installed modem-test. What should I try?

**Comment by miaszh** [comment]
Does the 3372-153 works fine? I'm looking around to find a confirmed working modem, and I cannot find any other model actually..
